scripts/tokenomics-emulation/emulations/add-nominator.emulation.ts

```diff
@@ -0,0 +1,59 @@
+import { IEmulation } from "./emulation.interface";
+import Network from "../network";
+import Accounts from "../accounts";
+import fs from "fs";
+import Nominator from "../nominators";
+
+class AddNominatorsEmulation implements IEmulation {
+  constructor(
+    private readonly account: Accounts,
+    private readonly networkConfig: any
+  ) {}
+
+  public async run(): Promise<void> {
+    const wsProvider = this.networkConfig.url;
+    const network = new Network(wsProvider, this.networkConfig.decimals);
+    await network.setup();
+    const nominator = new Nominator(network, this.account, this.networkConfig.decimals);
+    const validatorsCount = this.networkConfig.validators.amount;
+    const validators = await nominator.fetchValidators(validatorsCount);
+    const nominatorsCount = this.networkConfig.nominators.amount;
+    let validatorTurn: number = 0;
+
+    for (let i = 0; i < nominatorsCount; i++) {
+      console.log(`validatorTurn ${validatorTurn}`);
+      const validatorArray = [validators[validatorTurn]];
+      const stashAccountFile = fs.readFileSync(
+        `../generate-accounts/accounts/all/nominator-${i + 1}-stash`
+      );
+      const controllerAccountFile = fs.readFileSync(
+        `../generate-accounts/accounts/all/nominator-${i + 1}-controller`
+      );
+      const stashAccountMnemonic = JSON.parse(stashAccountFile.toString())
+        .mnemonic;
+      const controllerAccountMnemonic = JSON.parse(
+        controllerAccountFile.toString()
+      ).mnemonic;
+
+      const stashBond = this.networkConfig.nominators.stash_stake;
+      const exsistentialDeposit = await network.existentialDeposit();
+      const actualBondValue = stashBond - exsistentialDeposit;
+
+      await nominator.loadAccounts(
+        stashAccountMnemonic,
+        controllerAccountMnemonic
+      );
+      await nominator.addNominator(actualBondValue);
+      await nominator.setController();
+      await nominator.nominate(validatorArray);
+
+      if (i >= validatorsCount - 1) {
```

Comment:
You can use `(validatorTurn++) % validatorsCount`.

Reply:
It will assign the values backwards.

Reply:
It will be the same :-)

Reply:
Any updates here?

Reply:
Updated the logic